tests/rptest/services/direct_consumer_verifier.py

```diff
@@ -0,0 +1,308 @@
+# Copyright 2020 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from ducktape.tests.test import TestContext
+from ducktape.services.service import Service
+from ducktape.utils.util import wait_until
+import requests
+from dataclasses import dataclass, asdict, fields
+from typing import List, Optional, Dict, Any
+from enum import Enum
+import json
+
+from rptest.services.redpanda import AnyRedpandaService, RedpandaService, RedpandaServiceCloud
+from rptest.services.redpanda_installer import RedpandaInstaller
+
+
+def as_dict(obj):
+    """
+    Convert a dataclass instance to a dictionary, excluding None values.
+    Handles nested dataclasses, lists, and enums properly.
+    """
+    if hasattr(obj, '__dataclass_fields__'):
+        # Handle dataclass objects
+        result = {}
+        for field in fields(obj):
+            value = getattr(obj, field.name)
+            if value is not None:
+                result[field.name] = as_dict(value)
+        return result
+    elif isinstance(obj, list):
+        # Handle lists recursively
+        return [as_dict(item) for item in obj if item is not None]
+    elif hasattr(obj, 'value'):
+        # Handle enum objects
+        return obj.value
+    else:
+        # Return primitive values as-is
+        return obj
+
+
+class OffsetResetPolicy(Enum):
+    EARLIEST = 0
+    LATEST = 1
+
+
+class IsolationLevel(Enum):
+    READ_UNCOMMITTED = 0
+    READ_COMMITTED = 1
+
+
+@dataclass
+class BrokerAddress:
+    host: str
+    port: int
+
+
+@dataclass
+class DirectConsumerConfiguration:
```

Comment:
FYI I am working on generating protos and RPC clients for ducktape, this is fine for here however.